andela/ah-centauri-frontend
# #164829160 user should be able to signup/register

#### What does this PR do?
To be able to use the application and access authenticated routes, Users should be able to Signup and or Register using  Sign up forms on the signup route.

#### Description of Tasks to be completed?
1. The sign-up form should present to the user **descriptive** validation errors in case of any.
2. The user should receive a verification link on email upon successful registration.
3.  user should be able to sign up upon passes appropriate signup parameters
 
#### How should this be manually tested?
- Clone the project: git clone https://github.com/andela/ah-centauri-frontend
- cd into the project folder
- checkout to branch `feature/164829160-signup-or-register`
- Install dependencies: npm install
- Launch development server: npm start
- navigate to the signup page on the top right navigation bar
- enter a username, email, and password and click on the register
- a success message is displayed to confirm the email is yours.

#### Any background context you want to provide?
#### What are the relevant Pivotal Tracker Stories?
[#164829160](https://www.pivotaltracker.com/n/projects/2321088/#164829160)

#### Screenshots(if appropriate)
*figure shows Register Page*
![image](https://user-images.githubusercontent.com/33542127/57285864-cef78000-70bc-11e9-85de-7774c7066c24.png)
*figure shows Register Page shows error on submit empty form*
![image](https://user-images.githubusercontent.com/33542127/57285975-036b3c00-70bd-11e9-93c2-604625ac3c21.png)
*figure shows Register Page shows success on submit filled form*
![image](https://user-images.githubusercontent.com/33542127/57286001-12ea8500-70bd-11e9-9f61-cc9810bd14e7.png)
